python/src/otel/otel_sdk/opentelemetry/instrumentation/aws_lambda/__init__.py

```diff
@@ -85,35 +96,50 @@ def _uninstrument(self, **kwargs):
             self._wrapped_function_name,
         )
 
-    def _functionPatch(self, original_func, instance, args, kwargs):
-        lambda_context = args[1]
-        ctx_aws_request_id = lambda_context.aws_request_id
-        ctx_invoked_function_arn = lambda_context.invoked_function_arn
-        orig_handler = os.environ.get("ORIG_HANDLER", os.environ.get("_HANDLER"))
+
+def _instrument(tracer, wrapped_module_name, wrapped_function_name):
+    def _instrumented_lambda_handler_call(call_wrapped, instance, args, kwargs):
+        orig_handler_name = ".".join(
+            [wrapped_module_name, wrapped_function_name]
+        )
 
         # TODO: enable propagate from AWS by env variable
         xray_trace_id = os.environ.get("_X_AMZN_TRACE_ID", "")
-
-        lambda_name = os.environ.get("AWS_LAMBDA_FUNCTION_NAME")
-        function_version = os.environ.get("AWS_LAMBDA_FUNCTION_VERSION")
-
         propagator = AwsXRayFormat()
         parent_context = propagator.extract({"X-Amzn-Trace-Id": xray_trace_id})
 
-        with self._tracer.start_as_current_span(
-            name=orig_handler, context=parent_context, kind=SpanKind.SERVER
+        with tracer.start_as_current_span(
+            name=orig_handler_name, context=parent_context, kind=SpanKind.SERVER
         ) as span:
-            # Refer: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/faas.md#example
-            span.set_attribute("faas.execution", ctx_aws_request_id)
-            span.set_attribute("faas.id", ctx_invoked_function_arn)
-
-            # TODO: fix in Collector because they belong resource attrubutes
-            span.set_attribute("faas.name", lambda_name)
-            span.set_attribute("faas.version", function_version)
-
-            result = original_func(*args, **kwargs)
+            if span.is_recording():
```

Comment:
I am not sure it's necessary. When start the span and sampled, is_recording() is always true?

Reply:
That's a good question! Initially I didn't know why we did this and I just did it because we do it for every other instrumentation as seen in https://github.com/open-telemetry/opentelemetry-python/issues/1057, so I wanted `AwsLambdaInstrumentor` to be the same.

But now I understand that we need this because we don't know if the Span will be Sampled. It is started like you said but if the Sampler [doesn't sample this span](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L999) then we shouldn't record attributes.